### PR TITLE
Add online test for verify key attestation statement

### LINKED ISSUE #3: Update the test once our PKI server is ready
Do you think it would be worthwhile to update the test once our PKI server is ready?

_Originally posted by @xinyufort in https://github.com/fortanix/user-assurance/pull/2#discussion_r1317912303_

### PR DESCRIPTION
**Pull request by Taowyoo**
This PR fix #3

### CODE REVIEW

File: key-attestation/java-example/src/main/java/com/fortanix/keyattestationstatementverifier/Verify.java [hunk 1]
```diff
@@ -125,30 +132,42 @@ public static void verify(List<X509Certificate> authorityChain, X509Certificate
...
+                PKIXRevocationChecker.Option.NO_FALLBACK)); // don't fall back to OCSP checking
+
+        PKIXBuilderParameters pkixParams = new PKIXBuilderParameters(trustAnchors, new X509CertSelector());
+        pkixParams.setRevocationEnabled(verifyCrl);
+        if (verifyCrl){
+            pkixParams.addCertPathChecker(rc);
+        }
 
         // Validate CertPath
-        CertPathValidator validator = CertPathValidator.getInstance("PKIX", "BC");
```
Comment (Taowyoo) [inline key-attestation/java-example/src/main/java/com/fortanix/keyattestationstatementverifier/Verify.java:145]: Note: It turns out that `bouncycastle` provider does not support online CRL check.

File: key-attestation/java-example/src/test/java/com/fortanix/keyattestationstatementverifier/VerifyTest.java [hunk 2]
```diff
@@ -101,4 +110,88 @@ public void verifyStatementFullCheck() throws Exception {
...
+        String authString = "Basic " + appApiKeyString;
+        // Create a RSA key
+        String generateKeyUrl = FORTANIX_AMER_SAAS_SERVER_URL + "/crypto/v1/keys";
+        String newRsaKeyName = UUID.randomUUID().toString();
+        String generateKeyRequest = String.format(
+                "{\"name\":\"%s\",\"description\":\"\",\"obj_type\":\"RSA\",\"key_ops\":[\"APPMANAGEABLE\",\"SIGN\",\"VERIFY\"],\"key_size\":2048,\"pub_exponent\":65537,\"expirationDate\":null,\"enabled\":true,\"rsa\":{\"encryption_policy\":[{\"padding\":{\"OAEP\":{\"mgf\":{\"mgf1\":{}}}}}],\"signature_policy\":[{\"padding\":{\"PKCS1_V15\":{}}},{\"padding\":{\"PSS\":{\"mgf\":{\"mgf1\":{}}}}}]}}",
+                newRsaKeyName);
+        System.out.println(
+                String.format("Creating a new RSA key named '%s' through %s ...", newRsaKeyName, generateKeyUrl));
+        String generateKeyResponseString = sendHttpRequest(generateKeyUrl, "POST", generateKeyRequest, authString);
```
Comment (mzohreva) [inline key-attestation/java-example/src/test/java/com/fortanix/keyattestationstatementverifier/VerifyTest.java]: Why not use a proper DSM client for this?

Comment (Taowyoo) [reply key-attestation/java-example/src/test/java/com/fortanix/keyattestationstatementverifier/VerifyTest.java]: Fixed

File: key-attestation/java-example/src/main/java/com/fortanix/keyattestationstatementverifier/Verify.java [hunk 3]
```diff
@@ -125,30 +132,42 @@ public static void verify(List<X509Certificate> authorityChain, X509Certificate
...
-        // Set up PKIXParameters
-        PKIXParameters params = new PKIXParameters(trustAnchors);
-        params.setRevocationEnabled(verifyCrl);
+        CertPathBuilder cpb = CertPathBuilder.getInstance("PKIX");
+        PKIXRevocationChecker rc = (PKIXRevocationChecker) cpb.getRevocationChecker();
+        rc.setOptions(EnumSet.of(
+                PKIXRevocationChecker.Option.PREFER_CRLS, // prefer CLR over OCSP
+                PKIXRevocationChecker.Option.NO_FALLBACK)); // don't fall back to OCSP checking
+
+        PKIXBuilderParameters pkixParams = new PKIXBuilderParameters(trustAnchors, new X509CertSelector());
```
Comment (xinyufort) [inline key-attestation/java-example/src/main/java/com/fortanix/keyattestationstatementverifier/Verify.java:156]: Discussed with Yuxiang. It should be fine to also use `PKIXParameters(trustAnchors)` here too, but this is fine too

**Review by xinyufort — APPROVED** [review]
LGTM

**Pull request merged**